clang/test/CodeGenSYCL/group-local-memory.cpp

```diff
@@ -0,0 +1,44 @@
+// RUN: %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -S -emit-llvm %s -o - | FileCheck --check-prefixes=CHECK,CHECK-UNOPT %s
+// RUN: %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -S -emit-llvm %s -o - | FileCheck %s
```

Comment:
I think llvm/test/SYCLLowerIR/group_local_memory.ll should be enough to validate the pass. Just make sure it covers both optimized and unoptimized IR.
Testing a single pass by running the whole compilation stack seems like an overkill to me.
Are there any other reasons to have this test in addition to llvm/test/SYCLLowerIR/group_local_memory.ll?

Reply:
The main perpose of current check is to test that pass is executed during device code compilation in clang in both cases when llvm passes are turned on and turned off.

Reply:
There is much chipper way to achieve this with `-mllvm -debug-pass=Structure` - see https://github.com/intel/llvm/blob/sycl/clang/test/CodeGenCUDA/link-device-bitcode.cu#L37-L41 as an example.
Test program might be an empty file, if I'm not mistaken.